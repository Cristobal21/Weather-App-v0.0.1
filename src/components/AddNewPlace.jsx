import { useState } from "react"
import { Form } from 'react-bootstrap'
import PropTypes from 'prop-types'


export const AddNewPlace = ({ onNewPlace }) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = (e) => {
        e.preventDefault()
        if (inputValue.trim().length <= 1) return
        onNewPlace(inputValue.trim())
        setInputValue('')
    }
    
    return (

        <Form onSubmit={ onSubmit } aria-label="form" >
            <Form.Group>
                <Form.Control 
                    type="text" 
                    placeholder="Busca un lugar, paÃ­s o ciudad"
                    value={ inputValue }
                    onChange={ onInputChange }
                />
            </Form.Group>
        </Form>

    )
}

AddNewPlace.propTypes = {
    onNewPlace: PropTypes.func.isRequired,
}
