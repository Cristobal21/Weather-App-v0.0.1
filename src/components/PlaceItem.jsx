import { ListGroup } from "react-bootstrap"



export const PlaceItem = ({ name }) => {

    const handleClick = () => {
        //Hacer el redireccion a weathergrid o weatherdetail
        console.log('Hola hiciste click en > ', name);
    }

    return (
        <>
            <ListGroup>
                <ListGroup.Item action variant="info" className="mb-2" onClick={ handleClick }>
                    <p>{ name }</p>
                </ListGroup.Item>
            </ListGroup>
        </>
    )
}
