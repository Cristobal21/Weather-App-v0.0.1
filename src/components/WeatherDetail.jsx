import { Container, Row } from "react-bootstrap"
import { useFetchWeather } from "../hooks/useFetchWeather"

export const WeatherDetail = ({ lugar }) => {

    const { clima } = useFetchWeather( lugar )

    const dataWeather = () => {
        
    }


    return (
        <>
            <p>
                {
                    clima.map( data => data.name )
                }
            </p>
        </>
    )
}
