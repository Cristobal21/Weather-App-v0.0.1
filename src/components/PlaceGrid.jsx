import { useFetchPlaces } from "../hooks/useFetchPlaces"
import { PlaceItem } from "./PlaceItem"
import PropTypes from 'prop-types'


export const PlaceGrid = ({ lugar }) => {

    const { lugares } = useFetchPlaces( lugar )
    
    return (
        <>
            <p className="mt-3">{`Última búsqueda realizada: ${ lugar }`}</p>
            <hr />

            <div>
                {
                    lugares.map( (data) => (
                        <PlaceItem
                            key={ data.id }
                            { ...data }
                        />
                    ) )
                }
            </div>
        </>
    )
}


PlaceGrid.propTypes = {
    lugar: PropTypes.string.isRequired
}