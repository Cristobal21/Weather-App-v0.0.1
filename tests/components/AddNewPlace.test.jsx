import { fireEvent, render, screen } from "@testing-library/react"
import { AddNewPlace } from "../../src/components/AddNewPlace"

describe('Pruebas en <AddNewPlace/>', () => { 

    test('Debe hacer match con el snapshot', () => { 
        const { container } = render(<AddNewPlace onNewPlace={ () => {} } />)
        expect( container ).toMatchSnapshot()
    })

    test('Debe cambiar el valor de la caja de texto', () => { 
        render(<AddNewPlace onNewPlace={ () => {} } />)
        const input = screen.getByRole('textbox') // Hace referencia al elemento input que esta en el componente
        
        /** Para disparar un evento especifico */
        fireEvent.input( input, { target: { value: 'Buin' } } )
        expect( input.value ).toBe('Buin')
        // screen.debug()
    })

    test('Debe llamar al onNewPlace si el input tiene un valor', () => { 
        
        const inputValue = 'Buin'
        const onNewPlace = jest.fn()
        
        render(<AddNewPlace onNewPlace={ onNewPlace } />) // Se identifica al sujeto de pruebas

        const input = screen.getByRole('textbox') // Se hace referencia al elemento input
        const form = screen.getByRole('form') // Se hace referencia al elemento form

        fireEvent.input( input, { target: { value: inputValue } } ) // Se dispara el envento haciendo enfasis al valor que contiene el input
        fireEvent.submit( form ) // Se dispara el evento onsubmit del formulario
        // screen.debug()

        expect( input.value ).toBe('')

        expect( onNewPlace ).toHaveBeenCalled()
        expect( onNewPlace ).toHaveBeenCalledTimes(1)
        expect( onNewPlace ).toHaveBeenCalledWith( inputValue )
    })

    test('No debe de llamar el onNewPlace si el input esta vacío', () => {
        
        const onNewPlace = jest.fn()
        render(<AddNewPlace onNewPlace={ onNewPlace } />) // Se identifica al sujeto de pruebas

        const form = screen.getByRole('form')
        fireEvent.submit( form )

        expect( onNewPlace ).not.toHaveBeenCalled()
        
        // const input = screen.getByRole('textbox')
        // const form = screen.getByRole('form')

        // if( input.value.length > 1 ) {
        //     fireEvent.submit( form )
        //     expect( onNewPlace ).toHaveBeenCalledTimes(1)
        //     console.log('Hay algo en el input')
        // } else {
        //     expect( onNewPlace ).toHaveBeenCalledTimes(0)
        //     console.log('Input vacío');
        // }

    })

})