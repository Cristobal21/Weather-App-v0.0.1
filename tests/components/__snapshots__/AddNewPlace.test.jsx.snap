// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddNewPlace/> Debe hacer match con el snapshot 1`] = `
<div>
  <form
    aria-label="form"
    class=""
  >
    <div>
      <input
        class="form-control"
        placeholder="Busca un lugar, paÃ­s o ciudad"
        type="text"
        value=""
      />
    </div>
  </form>
</div>
`;
